Forge.TreeWalker.UnitTests/test/TreeSchemaValidatorTests.cs

```diff
@@ -0,0 +1,208 @@
+            Assert.AreEqual(true, res);
+            Assert.AreEqual(0, errorList.Count);
+        }
+
+        [TestMethod]
+        public void Test_ValidateSchemaFromPathWithCustomRulesInJSchema()
+        {
+            bool res = ForgeSchemaValidator.ValidateSchemaFromPath(tardigradeSchemaPath, jschemaRules, false, out IList<ValidationError> errorList);
+            Assert.AreEqual(true, res);
+            Assert.AreEqual(0, errorList.Count);
+        }
+
+        [TestMethod]
+        public void Test_ValidateSchemasFromDirectoryListWithCustomRulesInString()
+        {
+            bool res = ForgeSchemaValidator.ValidateSchemaFromDirectory(schemaDirectoryPath, stringRules, false, out IList<ValidationError> errorList);
+            Assert.AreEqual(true, res);
+            Assert.AreEqual(0, errorList.Count);
+        }
+
+        [TestMethod]
+        public void Test_ValidateSchemasFromDirectoryListWithCustomRulesInJSchema()
+        {
+            bool res = ForgeSchemaValidator.ValidateSchemaFromDirectory(schemaDirectoryPath, jschemaRules, false, out IList<ValidationError> errorList);
+            Assert.AreEqual(true, res); 
+            Assert.AreEqual(0, errorList.Count);
+        }
+
+        [TestMethod]
+        public void Test_ValidateInvalidSchemaInString()
+        {
+            bool res = ForgeSchemaValidator.ValidateSchemaAsString(tardigradeSchemaString, jschemaRules, false, out IList<ValidationError> errorList);
+            Assert.AreEqual(true, res);
+            Assert.AreEqual(0, errorList.Count);
+        }
+
+        [TestMethod]
+        public void Test_ValidateInvalidSchemaDirectoryPath()
+        {
+            try
+            {
+                ForgeSchemaValidator.ValidateSchemaFromDirectory(invalidSchemaDirectoryPath, jschemaRules, false, out IList<ValidationError> errorList);
+            }
+            catch (Exception e)
+            {
+                Assert.AreEqual("The directory name is invalid.\r\n", e.Message);
+            }
+        }
+
+        [TestMethod]
+        public void Test_ValidateSchemaWithErrorContent()
+        {
+            bool res = ForgeSchemaValidator.ValidateSchemaAsString(invalidSchemaWithErrorContent, jschemaRules, false, out IList<ValidationError> errorList);
+            Assert.AreEqual(false, res);
+            Assert.AreEqual("JSON is valid against no schemas from 'oneOf'.", errorList.First().Message);
+        }
+
+        [TestMethod]
+        public void Test_ValidateInvalidSchemaAsDictionary()
+        {
+            bool res = ForgeSchemaValidator.ValidateSchemaAsString(invalidSchemaWithErrorContent, jschemaRules, true, out IList<ValidationError> errorList);
+            Assert.AreEqual(false, res);
+            Assert.AreEqual("JSON is valid against no schemas from 'oneOf'.", errorList.First().Message);
+        }
+
+        [TestMethod]
+        public void Test_ValidateValidSchemaAsDictionary()
```

Comment:
> public void Test_ValidateValidSchemaAsDictionary() [](http://example.com/codeflow?start=7&length=51)

Let's organize the test methods to align with the ForgeSchemaValidator methods. I mean have them in the same order, testing ForgeTree, then ForgeTreeDictionary, string, path, directory.

Put the invalid tests at the end.